api/src/wfl/service/terra.clj

```diff
@@ -0,0 +1,53 @@
+(ns wfl.service.terra
+  "Analyze data in Terra using the Firecloud/Terra API."
+  (:require [clojure.data.json :as json]
+            [clojure.string :as str]
+            [clj-http.client :as http]
+            [wfl.once :as once]))
+
+(defn workspace-api-url
+  [terra-url workspace-namespace workspace-name]
+  (str terra-url "/api/workspaces/" workspace-namespace "/" workspace-name))
+
+(defn create-submission
+  "Submit samples in a workspace for analysis with a method configuration in Terra."
+  [terra-url workspace-namespace workspace-name method-configuration-name
+   method-configuration-namespace entity-type entity-name]
+  (let [workspace-url (workspace-api-url terra-url workspace-namespace workspace-name)]
+    (-> {:method    :post
+         :url       (str workspace-url "/submissions")
+         :content-type :application/json
+         :headers   (once/get-auth-header)
+         :body      (json/write-str {
+                    :methodConfigurationNamespace method-configuration-namespace
+                    :methodConfigurationName method-configuration-name
+                    :entityType entity-type
+                    :entityName entity-name
+                    :useCallCache true
+                   } :escape-slash false)}
+      http/request :body
+      (json/read-str :key-fn keyword)
+      :submissionId)))
+
+(defn get-submission
+  "Get information about a Terra Cromwell submission."
+  [terra-url workspace-namespace workspace-name submission-id]
+  (let [workspace-url (workspace-api-url terra-url workspace-namespace workspace-name)]
```

Comment:
Why not call `http/get` directly?

Reply:
I think there's some slight benefit to making the code here the same as that for create-submission. If there's a http/post we can call there then sure I guess but this is pretty readable to me as-is.

Reply:
No particular reason...I referenced another module when writing this one and it used `http/request` but I think `http/get` is a bit clearer!

Reply:
Oh I didn't see Jack's comment...I updated these functions to use `http/get` and `http/post`. Let me know if you all think this is more readable!